cdc/processor/sinkmanager/tasks.go

```diff
@@ -38,7 +38,7 @@ var (
 type writeSuccessCallback func(lastWrittenPos engine.Position)
 
 // Used to get an upper bound.
-type upperBoundGetter func() engine.Position
+type upperBoundGetter func(*tableSinkWrapper) engine.Position
```

Comment:
Could you please make this func be a method of task? Because it is wried to pass take sink back to this func, it already has table sink pointer in task struct. We can just add `getUpperBoundForSink` and `getUpperBoundForRedo` to the task struct to get the upper bound ts.

Reply:
It can be a pointer. Atomic pointer for barrier ts.